lib/concurrent/edge/lock_free_linked_set.rb

```diff
@@ -0,0 +1,131 @@
+require 'concurrent/edge/lock_free_linked_set/node'
+require 'concurrent/edge/lock_free_linked_set/window'
+
+module Concurrent
+  module Edge
+    class LockFreeLinkedSet
```

Comment:
Could you include short description of the class and if possible a link to your source of the algorithm?